tools/tsp-client/src/fs.ts

```diff
@@ -47,3 +48,48 @@ export async function tryReadTspLocation(rootDir: string): Promise<string | unde
   }
   return undefined;
 }
+
+export async function readTspLocation(rootDir: string): Promise<[string, string, string, string[]]> {
+  try {
+    const yamlPath = path.resolve(rootDir, "tsp-location.yaml");
+    const fileStat = await stat(yamlPath);
+    if (fileStat.isFile()) {
+      const fileContents = await readFile(yamlPath, "utf8");
+      const locationYaml = parseYaml(fileContents);
+      var { directory, commit, repo, additionalDirectories } = locationYaml;
+      if (!directory || !commit || !repo) {
+        throw new Error("Invalid tsp-location.yaml");
+      }
+      Logger.info(`Additional directories: ${additionalDirectories}`)
+      if (!additionalDirectories) {
+        additionalDirectories = [];
+      }
+      return [ directory, commit, repo, additionalDirectories ];
+    }
+  } catch (e) {
+    Logger.error(`Error reading tsp-location.yaml: ${e}`);
+  }
+  throw new Error("Could not find tsp-location.yaml");
+}
+
+
+export async function findEmitterPackage(emitterPath: string): Promise<string | undefined> {
+  if (existsSync(emitterPath)) {
+    const emitter = await readFile(emitterPath, 'utf8').then(data => {
+      const obj = JSON.parse(data);
+      if (!obj || !obj.dependencies) {
+        throw new Error("Invalid emitter-package.json");
+      }
+      var languages: string[] = ["@azure-tools/typespec-csharp", "@azure-tools/typespec-java", "@azure-tools/typespec-ts", "@azure-tools/typespec-python", "@typespec/openapi3"];
+      for (var lang in languages) {
+        if (obj.dependencies[languages[lang]!]) {
```

Comment:
instead of checking a hardcoded list, could we reduce the dependencies array to look for something that starts with `@azure-tools/typespec-` and take the first one? I suppose we'd still have to handle openapi3 specially, but otherwise it seems more extensible